cmd/coap/main.go

```diff
@@ -58,14 +61,23 @@ func main() {
 	defer tcHandler.Close()
 	logger.Info("Successfully connected to things grpc server " + tcHandler.Secure())
 
+	tracer, traceCloser, err := jaegerClient.NewTracer(svcName, cfg.JaegerURL)
+	if err != nil {
+		logger.Fatal(fmt.Sprintf("failed to init Jaeger: %s", err))
+	}
+	defer traceCloser.Close()
+
 	nps, err := brokers.NewPubSub(cfg.BrokerURL, "", logger)
 	if err != nil {
 		logger.Fatal(fmt.Sprintf("failed to connect to message broker: %s", err))
 	}
+	nps = pstracing.NewPubSub(nps, tracer)
 	defer nps.Close()
 
 	svc := coap.New(tc, nps)
 
+	svc = tracing.New(tracer, svc)
```

Comment:
This is not related to this particular PR, but please check the param order in `tracing.New` and make it consistent for all tracing middlewares: It's always `(svc, tracer)` pair, but we're not consistent with the order. Let's make it always `(tracer, svc)`.